web/tools.html

```diff
@@ -95,5 +103,44 @@ <h4>Tools</h4>
             })
             .catch(error => console.error('Error loading footer:', error));
         </script>
+
+        <script>
+            document.addEventListener('DOMContentLoaded', function() {
+                const manifestUrl = document.getElementById('manifestUrl');
+                const loadManifest = document.getElementById('loadManifest');
+                const manifestMessage = document.getElementById('manifestMessage');
+
+                loadManifest.addEventListener('click', function() {
+                    const url = manifestUrl.value.trim();
+                    if (!url) {
+                        manifestMessage.textContent = 'Please enter a URL.';
+                        manifestMessage.style.color = 'red';
+                        return;
+                    }
+
+                    manifestMessage.textContent = 'Loading...';
+                    manifestMessage.style.color = 'black';
+
+                    fetch(url)
+                        .then(response => {
+                            if (!response.ok) {
+                                throw new Error('Network response was not ok');
+                            }
+                            return response.json();
+                        })
+                        .then(data => {
+                            // process JSON URL data
```

Comment:
I feel like you could perform JSON structure validation here to ensure that it's a valid manifest before proceeding